subprojects/common/http-client/src/main/kotlin/com/avito/http/HttpClientProvider.kt

```diff
@@ -0,0 +1,72 @@
+package com.avito.http
+
+import com.avito.android.stats.CountMetric
+import com.avito.android.stats.SeriesName
+import com.avito.android.stats.StatsDSender
+import com.avito.http.TryFailCallback.Companion.combine
+import com.avito.http.internal.ServiceMetricsInterceptor
+import com.avito.http.internal.StatsdServiceEventsListener
+import com.avito.logger.LoggerFactory
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import java.util.concurrent.TimeUnit
+
+public class HttpClientProvider(
+    private val statsDSender: StatsDSender,
+    private val loggerFactory: LoggerFactory,
+    private val builderTransform: OkHttpClient.Builder.() -> OkHttpClient.Builder = { this }
+) {
+
+    private val builder = OkHttpClient.Builder()
+
+    public fun provide(
+        serviceName: String,
+        timeoutMs: Long? = null,
+        retryPolicy: RetryPolicy? = null,
+        builderTransform: OkHttpClient.Builder.() -> OkHttpClient.Builder = { this }
+    ): OkHttpClient {
+
+        val seriesName = SeriesName.create("service", serviceName)
+
+        return builder
+            .also { this.builderTransform(it) }
+            .also { builderTransform(it) }
```

Comment:
What client is using both transformations? Is one is not enough?

Reply:
One is enough from API perspective, but constructor one needed for testing purposes;
i'll try to get rid of it